src/fsharp/build.fs

```diff
@@ -2705,9 +2705,15 @@ type TcConfig private (data : TcConfigBuilder,validate:bool) =
         let isNetModule = String.Compare(ext,".netmodule",StringComparison.OrdinalIgnoreCase)=0 
         if String.Compare(ext,".dll",StringComparison.OrdinalIgnoreCase)=0 
            || String.Compare(ext,".exe",StringComparison.OrdinalIgnoreCase)=0 
-           || isNetModule then 
+           || isNetModule then
 
-            let resolved = TryResolveFileUsingPaths(tcConfig.SearchPathsForLibraryFiles,m,nm)
+            let searchPaths =
+                if m <> range0 && m <> rangeStartup && m <> rangeCmdArgs && FileSystem.IsPathRootedShim m.FileName then
```

Comment:
It's to ensure the modified resolution paths are only used for explicit `#r` references.

Real `#r` references will carry along a proper range pointing at the declaring source file, which is what we are interested in.

Other types of references (from command line, auto-added at startup, etc) are given well-known dummy ranges (`range0`, `rangeStartup`, `rangeCmdArgs`), so we just filter those out.

The `IsPathRooted` check might not be strictly needed, but that path will end up being used as the root of a potential resolution directory, so I think it's prudent to double-check.